hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java

```diff
@@ -181,18 +181,19 @@ public void flushMaps() {
     public void destroyMap(String mapName) {
         final PartitionContainer[] containers = partitionContainers;
         final List<Future> futures = new ArrayList<Future>(containers.length);
+        Address localAddress = getNodeEngine().getLocalMember().getAddress();
         for (PartitionContainer container : containers) {
             if (container != null) {
```

Comment:
To prevent excessive nesting, I normally do a if(container == null) continue;

Jump out of something as soon as you have no reason to be there.